src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -569,7 +569,7 @@ public DatabaseSearchResult TryGetOrCreateDatabase(StringSegment databaseName)
                 if (databaseRecord.Shards?.Length > 0)
                 {
                     var shardedContext = new ShardedContext(_serverStore, databaseRecord);
-                    _shardedDatabases.GetOrAdd(databaseName, Task.FromResult(shardedContext));
+                    shardedContext = _shardedDatabases.GetOrAdd(databaseName, Task.FromResult(shardedContext)).Result;
```

Comment:
Is is possible that the value here would be an async task?

Reply:
No since the only way we add a result to this dictionary is using Task.FromResult